pkg/ssh/ssh.go

```diff
@@ -502,42 +494,253 @@ func RunSSHUploadStakingFiles(host *models.Host, nodeInstanceDirPath string) err
+	}
+	var remoteNodeConfig map[string]interface{}
+	if err := json.Unmarshal(remoteNodeConfigBytes, &remoteNodeConfig); err != nil {
+		return fmt.Errorf("error unmarshalling remote node config: %w", err)
+	}
+	subnetNodeConfigBytes, err := os.ReadFile(subnetNodeConfigPath)
+	if err != nil {
+		return fmt.Errorf("error reading subnet node config: %w", err)
+	}
+	var subnetNodeConfig map[string]interface{}
+	if err := json.Unmarshal(subnetNodeConfigBytes, &subnetNodeConfig); err != nil {
+		return fmt.Errorf("error unmarshalling subnet node config: %w", err)
+	}
+	mergedNodeConfig := utils.MergeJSONMaps(remoteNodeConfig, subnetNodeConfig)
+	mergedNodeConfigBytes, err := json.MarshalIndent(mergedNodeConfig, "", " ")
+	if err != nil {
+		return fmt.Errorf("error creating merged node config: %w", err)
+	}
+	if err := os.WriteFile(tmpFile.Name(), mergedNodeConfigBytes, constants.WriteReadUserOnlyPerms); err != nil {
+		return err
+	}
+	return host.Upload(tmpFile.Name(), remoteconfig.GetRemoteAvalancheNodeConfig(), constants.SSHFileOpsTimeout)
+}
+
+// RunSSHSyncSubnetData syncs subnet data required
+func RunSSHSyncSubnetData(app *application.Avalanche, host *models.Host, network models.Network, subnetName string) error {
+	sc, err := app.LoadSidecar(subnetName)
+	if err != nil {
+		return err
+	}
+	subnetID := sc.Networks[network.Name()].SubnetID
+	if subnetID == ids.Empty {
+		return errors.New("subnet id is empty")
+	}
+	subnetIDStr := subnetID.String()
+	blockchainID := sc.Networks[network.Name()].BlockchainID
+	// genesis config
+	genesisFilename := filepath.Join(app.GetNodesDir(), host.GetCloudID(), constants.GenesisFileName)
+	if err := host.Upload(genesisFilename, remoteconfig.GetRemoteAvalancheGenesis(), constants.SSHFileOpsTimeout); err != nil {
+		return fmt.Errorf("error uploading genesis config to %s: %w", remoteconfig.GetRemoteAvalancheGenesis(), err)
+	}
+	// end genesis config
+	// subnet node config
+	subnetNodeConfigPath := app.GetAvagoNodeConfigPath(subnetName)
+	if utils.FileExists(subnetNodeConfigPath) {
+		if err := mergeSubnetNodeConfig(host, subnetNodeConfigPath); err != nil {
+			return err
+		}
+	}
+	// subnet config
+	if app.AvagoSubnetConfigExists(subnetName) {
+		subnetConfig, err := app.LoadRawAvagoSubnetConfig(subnetName)
+		if err != nil {
+			return fmt.Errorf("error loading subnet config: %w", err)
+		}
+		subnetConfigFile, err := os.CreateTemp("", "avalanchecli-subnet-*.json")
+		if err != nil {
+			return err
+		}
+		defer os.Remove(subnetConfigFile.Name())
+		if err := os.WriteFile(subnetConfigFile.Name(), subnetConfig, constants.WriteReadUserOnlyPerms); err != nil {
+			return err
+		}
+		subnetConfigPath := filepath.Join(constants.CloudNodeConfigPath, "subnets", subnetIDStr+".json")
+		if err := host.MkdirAll(filepath.Dir(subnetConfigPath), constants.SSHDirOpsTimeout); err != nil {
+			return err
+		}
+		if err := host.Upload(subnetConfigFile.Name(), subnetConfigPath, constants.SSHFileOpsTimeout); err != nil {
```

Comment:
can we have an upload that takes as input a slice of bytes, so we dont need to create tmp files?

Reply:
agree. we should have another func for this